rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,267 @@
+
+This approach would extend the [RFC 2064](https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md#collecting-uniform-context-data) by _just_ adding `pipeline_id` and `pipeline_version` to the context.
+
+When [spawning a transform](https://github.com/timberio/vector/blob/v0.15.0/src/topology/mod.rs#L574), adding the optional pipeline information to the span will populate the metrics.
+
+```rust
+let span = error_span!(
+    "transform",
+    component_kind = "transform",
+    component_name = %task.name(),
+    component_type = %task.typetag(),
+    pipeline_id = %task.pipeline_id(),
+    pipeline_version = %task.pipeline_version(),
+);
+```
+
+## Doc-level Proposal
+
+With this pipeline definition, the following `vector.toml`
+
+```toml
+[sources.docker]
+type = "docker_logs"
+...
+
+[transforms.nginx_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter nginx logs from docker logs
+'''
+
+[transforms.nginx]
+input = ["nginx_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the nginx logs
+'''
+
+[transforms.apache_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter apache logs from docker logs
+'''
+
+[transforms.apache]
+input = ["apache_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the apache logs
+'''
+
+[sinks.ouput]
+inputs = ["apache", "nginx"]
+type = "console"
+...
+```
+
+can be split into the following `pipelines/frontend.toml`
+
+```toml
+[transforms.nginx_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter nginx logs from docker logs
+'''
+
+[transforms.nginx]
+input = ["frontend.nginx_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the nginx logs
+'''
+
+[transforms.apache_filter]
+input = ["docker"]
+type = "remap"
+source = '''
+  # do something that filter apache logs from docker logs
+'''
+
+[transforms.apache]
+input = ["frontend.apache_filter"]
+type = "remap"
+source = '''
+  # do some transformations with the apache logs
+'''
+```
+
+and `vector.toml`
+
+```toml
+[sources.docker]
+type = "docker_logs"
+...
+
+[sinks.ouput]
+inputs = ["frontend.apache", "frontend.nginx"]
+type = "console"
+...
+```
+
+## Rationale
+
+- This improves the readability of the `vector.toml`.
+
+## Drawbacks
+
+_TODO_
+
+## Alternatives
+
+- Do nothing: we can already use several configuration files, people could split their existing configuration.
+
+This would imply some duplication if a transform is used in multiple configuration files.
+
+- Evolve vector to use a tag/filter model like our competitors, have a 'pipeline' be a 'tag'.
+
+
+- Run a single vector per-'pipeline' and support metric tagging to distinguish at the telemetry level.
+
+
+## Outstanding Questions
+
+- Should the `pipelines` directory location be configurable through the cli?
```

Comment:
I would make these sub-headings with discussion below. That helps with reading and commenting. See: https://github.com/timberio/vector/tree/master/rfcs#use-good-technical-grammar

Reply:
In addition, we need a lot more explanation and examples for some of these, I don't think others will understand them otherwise.